lightning/src/ln/peer_handler.rs

```diff
@@ -1272,4 +1284,124 @@ mod tests {
 		peers[0].timer_tick_occured();
 		assert_eq!(peers[0].peers.lock().unwrap().peers.len(), 0);
 	}
+
+	pub struct TestRoutingMessageHandler {}
+
+	impl TestRoutingMessageHandler {
+		pub fn new() -> Self {
+			TestRoutingMessageHandler {}
+		}
+
+	}
+	impl msgs::RoutingMessageHandler for TestRoutingMessageHandler {
+		fn handle_node_announcement(&self, _msg: &msgs::NodeAnnouncement) -> Result<bool, msgs::LightningError> {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_channel_announcement(&self, _msg: &msgs::ChannelAnnouncement) -> Result<bool, msgs::LightningError> {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_channel_update(&self, _msg: &msgs::ChannelUpdate) -> Result<bool, msgs::LightningError> {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_htlc_fail_channel_update(&self, _update: &msgs::HTLCFailChannelUpdate) {}
+		fn get_next_channel_announcements(&self, starting_point: u64, batch_amount: u8) -> Vec<(msgs::ChannelAnnouncement, msgs::ChannelUpdate,msgs::ChannelUpdate)> {
+			let mut chan_anns = Vec::new();
+			let end: u64 = if starting_point > 0 { 6 } else { batch_amount.into() };
```

Comment:
Maybe staring_point + batch_amount.into() - 1, to avoid any assumptions about the batch_amount?